### PR TITLE
Setup initial integration tests

### PR DESCRIPTION
**Pull request by pradtke**
* Restore old www path names in routes to ease upgrades
* Make wctx optional since it was not required in the released version.

### CODE REVIEW

**Comment by tvdijen** [comment]
Great work Patrick!  My only suggestion would be to use the PEMCertificatesMock instead of adding the sample certs to this repo.
Here's an example;  https://github.com/simplesamlphp/saml2/blob/90ddb45233c6abd79c3a912d55f715519f7c9352/tests/SAML2/SignedElementTestTrait.php#L54

**Comment by pradtke** [comment]
Thanks for pointing out PEMCertificatesMock @tvdijen! That's a good change.
I've updated the PR.

**Comment by tvdijen** [comment]
One more question I'm asking myself, without checking too much of the code, is whether we still need the BuiltinServer when we have routing? The PRP-test you're adding seems a lot like a controller-test, but this one is not covering the AdfsController-class? Or should we add a @covers tag to see the added coverage?

File: tests/www/PrpIntegrationTest.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+declare(strict_types=1);
+
+namespace Simplesamlphp\adfs;
+
+use PHPUnit\Framework\TestCase;
+use SimpleSAML\TestUtils\BuiltInServer;
+
+class PrpIntegrationTest extends TestCase
+{
+
```
Comment (tvdijen) [inline tests/www/PrpIntegrationTest.php:12]: Nitpick, but this newline breaks PSR-12 compliancy

File: composer.json [hunk 2]
```diff
@@ -29,11 +29,13 @@
...
         "simplesamlphp/composer-module-installer": "^1.1.7",
-        "simplesamlphp/assert": "^0.1.0"
+        "simplesamlphp/assert": "^0.1.0",
+        "simplesamlphp/xml-security": "~0.0.10"
     },
     "require-dev": {
         "simplesamlphp/simplesamlphp": "dev-master",
-        "simplesamlphp/simplesamlphp-test-framework": "^1.0.4"
+        "simplesamlphp/simplesamlphp-test-framework": "^1.0.4",
+        "ext-curl": "*"
```
Comment (tvdijen) [inline composer.json:38]: I don't see curl being used in this PR?  Was it missing before?

Comment (pradtke) [reply composer.json:38]: The test reference things like `CURLOPT_FOLLOWLOCATION`. and PhpStorm flagged it as something to include in `composer.json`.  Adding it was really just me making warnings go away, so it is something that we can remove if you think that's best.

File: tests/www/PrpIntegrationTest.php (hunk 1 above)
Comment (pradtke) [reply tests/www/PrpIntegrationTest.php:12]: Thanks. When I run `./vendor/bin/phpcs tests/www/PrpIntegrationTest.php ` it didn't flag that. Any idea why not?  (I'll push a change with newline removed)

**Comment by pradtke** [comment]
> One more question I'm asking myself, without (I admit) checking too much of the code, is whether we still need the BuiltinServer when we have routing?

I was hoping I wouldn't. But as `ADFS::receiveAuthnRequest` does it's thing eventually `IdP::postAuthProc` is called which expects the `Responder` function to have called `exit()`. So as part of this PR `ADFS::postResponse` now calls `exit`. Without that `exit` the `Assert::true(false);` in `postAuthProc` would fail.

File: tests/www/PrpIntegrationTest.php (hunk 1 above)
Comment (tvdijen) [reply tests/www/PrpIntegrationTest.php]: I wouldn't know to be honest... Maybe they loosened the specs?  I'd be glad to see the newline go to be consistent

File: composer.json (hunk 2 above)
Comment (tvdijen) [reply composer.json:38]: Not it's okay if we use it!  Missed it

**Comment by tvdijen** [comment]
Superseded by #10

**Pull request closed without merging**